### ISSUE TITLE
Receive notifications on Background and when app is killed

### ISSUE THREAD

**Issue by xellDart** [peerwaya/flutter_voip_push_notification#2]
Hi, please help em, all is working on foreground, but when app is on Background method onResume  method kill app, another question, I need show notification alert when app is killed, how I can do this?

**Comment by VictorUvarov** [comment]
This plugin does not work for normal notifications. When using PushKit apps must report a system call otherwise iOS will kill your app after a while.

**Comment by vuitv** [comment]
i has same issues

**Comment by savadmv** [comment]
> This plugin does not work for normal notifications. When using PushKit apps must report a system call otherwise iOS will kill your app after a while.

How to test app with VoIP push . I want move further only if it is working . Is there any website or something to test this. ..?

**Comment by ziasultan2** [comment]
> This plugin does not work for normal notifications. When using PushKit apps must report a system call otherwise iOS will kill your app after a while.

I want to show notification like uber driver notification. What I need to do in this case. Please give enlighten me with some details

**Comment by savadmv** [comment]
I had same issue and solved it 

```
import UIKit
import PushKit
import flutter_voip_push_notification
import flutter_call_kit
import CallKit

import Flutter


@UIApplicationMain
@objc class AppDelegate: FlutterAppDelegate ,PKPushRegistryDelegate{
    
    func pushRegistry(_ registry: PKPushRegistry, didUpdate pushCredentials: PKPushCredentials, for type: PKPushType) {
             NSLog("didUpdatePushCredentials: \(pushCredentials.token)")
             // Process the received push
             FlutterVoipPushNotificationPlugin.didUpdate(pushCredentials, forType: type.rawValue);
         }
         
         func pushRegistry(_ registry: PKPushRegistry,
                           didReceiveIncomingPushWith payload: PKPushPayload,
                           for type: PKPushType,
                           completion: @escaping () -> Void) {
             let uuid = NSUUID().uuidString
             let handle = "+819011111111"
             let callerName = "Robert"

             FlutterCallKitPlugin.reportNewIncomingCall(uuid, handle: handle, handleType: "generic", hasVideo: false, localizedCallerName: callerName, fromPushKit: true)
             // Process the received push
             FlutterVoipPushNotificationPlugin.didReceiveIncomingPush(with: payload, forType: type.rawValue)

             completion()
         }
    
}
```

Add this code in your AppDelegate.swift 
Send Voip Notification from server . 
If your backend works in Nod Js  , fine solution [here](https://stackoverflow.com/questions/32472859/im-trying-to-use-apns-library-for-node-js-and-get-an-error-typeerror-cannot/62172188#62172188
)

**Comment by ziasultan2** [comment]
Thank you for your solution.  Is there any way I can pass caller name send from voip push notification to show on callkit

> I had same issue and solved it
> 
> ```
> import UIKit
> import PushKit
> import flutter_voip_push_notification
> import flutter_call_kit
> import CallKit
> 
> import Flutter
> 
> 
> @UIApplicationMain
> @objc class AppDelegate: FlutterAppDelegate ,PKPushRegistryDelegate{
>     
>     func pushRegistry(_ registry: PKPushRegistry, didUpdate pushCredentials: PKPushCredentials, for type: PKPushType) {
>              NSLog("didUpdatePushCredentials: \(pushCredentials.token)")
>              // Process the received push
>              FlutterVoipPushNotificationPlugin.didUpdate(pushCredentials, forType: type.rawValue);
>          }
>          
>          func pushRegistry(_ registry: PKPushRegistry,
>                            didReceiveIncomingPushWith payload: PKPushPayload,
>                            for type: PKPushType,
>                            completion: @escaping () -> Void) {
>              let uuid = NSUUID().uuidString
>              let handle = "+819011111111"
>              let callerName = "Robert"
> 
>              FlutterCallKitPlugin.reportNewIncomingCall(uuid, handle: handle, handleType: "generic", hasVideo: false, localizedCallerName: callerName, fromPushKit: true)
>              // Process the received push
>              FlutterVoipPushNotificationPlugin.didReceiveIncomingPush(with: payload, forType: type.rawValue)
> 
>              completion()
>          }
>     
> }
> ```
> 
> Add this code in your AppDelegate.swift
> Send Voip Notification from server .
> If your backend works in Nod Js , fine solution [here](https://stackoverflow.com/questions/32472859/im-trying-to-use-apns-library-for-node-js-and-get-an-error-typeerror-cannot/62172188#62172188)

**Comment by savadmv** [comment]
I don’t think you can that with this library . But it my be possible my modifying it. My requirement was to show a fixed Name . so i didn’t try for that   .

**Comment by ziasultan2** [comment]
Thank you dear for your reply

**Comment by VictorUvarov** [comment]
@ziasultan2 Yes you can get all the info from the payload that you sent over from your server

Overview
- when you recieve a voip notification parse the notification
- if the notification is good, tell callkit to start the call
- you can fill in the call details from the payload you get from the server

#### Objective-C
```objective-c
NSString *localizedName = payload[@"identifier"];
```

#### Swift
```swift
guard let localizedName = payload["identifier"] as? String else { return }
```

I would look at VoxImplants [CallKit](https://github.com/voximplant/flutter_callkit) README for more details

**Comment by Suhailakl** [comment]
> I had same issue and solved it
> 
> ```
> import UIKit
> import PushKit
> import flutter_voip_push_notification
> import flutter_call_kit
> import CallKit
> 
> import Flutter
> 
> 
> @UIApplicationMain
> @objc class AppDelegate: FlutterAppDelegate ,PKPushRegistryDelegate{
>     
>     func pushRegistry(_ registry: PKPushRegistry, didUpdate pushCredentials: PKPushCredentials, for type: PKPushType) {
>              NSLog("didUpdatePushCredentials: \(pushCredentials.token)")
>              // Process the received push
>              FlutterVoipPushNotificationPlugin.didUpdate(pushCredentials, forType: type.rawValue);
>          }
>          
>          func pushRegistry(_ registry: PKPushRegistry,
>                            didReceiveIncomingPushWith payload: PKPushPayload,
>                            for type: PKPushType,
>                            completion: @escaping () -> Void) {
>              let uuid = NSUUID().uuidString
>              let handle = "+819011111111"
>              let callerName = "Robert"
> 
>              FlutterCallKitPlugin.reportNewIncomingCall(uuid, handle: handle, handleType: "generic", hasVideo: false, localizedCallerName: callerName, fromPushKit: true)
>              // Process the received push
>              FlutterVoipPushNotificationPlugin.didReceiveIncomingPush(with: payload, forType: type.rawValue)
> 
>              completion()
>          }
>     
> }
> ```
> 
> Add this code in your AppDelegate.swift Send Voip Notification from server . If your backend works in Nod Js , fine solution [here](https://stackoverflow.com/questions/32472859/im-trying-to-use-apns-library-for-node-js-and-get-an-error-typeerror-cannot/62172188#62172188)

The application not crashing for me but no notification receiving in the background.But,It works in foreground.

**Comment by evicoach** [comment]
> here
App crashed with error message below
reason: '-[CXProviderConfiguration initWithLocalizedName:]: parameter 'localizedName' cannot be nil'
